src/Engine/Data/Mesh.hpp

```diff
@@ -412,6 +413,101 @@ class RA_ENGINE_API PolyMesh : public IndexedGeometry<Core::Geometry::PolyMesh>
     Core::AlignedStdVector<IndexType> m_triangleIndices;
 };
 
+/// create a TriangleMesh, PolyMesh or other Core::*Mesh from GeometryData
+template <typename CoreMeshType>
+CoreMeshType createCoreMeshFromGeometryData( const Ra::Core::Asset::GeometryData* data ) {
+    CoreMeshType mesh;
+    typename CoreMeshType::PointAttribHandle::Container vertices;
+    typename CoreMeshType::NormalAttribHandle::Container normals;
+    typename CoreMeshType::IndexContainerType indices;
+
+    vertices.reserve( data->getVerticesSize() );
+    std::copy(
+        data->getVertices().begin(), data->getVertices().end(), std::back_inserter( vertices ) );
+
+    if ( data->hasNormals() )
+    {
+        normals.reserve( data->getVerticesSize() );
+        std::copy(
+            data->getNormals().begin(), data->getNormals().end(), std::back_inserter( normals ) );
+    }
+
+    const auto& faces = data->getFaces();
+    indices.reserve( faces.size() );
+    std::copy( faces.begin(), faces.end(), std::back_inserter( indices ) );
+    mesh.setVertices( std::move( vertices ) );
+    mesh.setNormals( std::move( normals ) );
+
+    if ( data->hasTangents() )
+    {
+        mesh.addAttrib( Data::Mesh::getAttribName( Data::Mesh::VERTEX_TANGENT ),
+                        data->getTangents() );
+    }
+
+    if ( data->hasBiTangents() )
+    {
+        mesh.addAttrib( Data::Mesh::getAttribName( Data::Mesh::VERTEX_BITANGENT ),
+                        data->getBiTangents() );
+    }
+
+    if ( data->hasTextureCoordinates() )
+    {
+        mesh.addAttrib( Data::Mesh::getAttribName( Data::Mesh::VERTEX_TEXCOORD ),
+                        data->getTexCoords() );
+    }
+
+    if ( data->hasColors() )
+    {
+        mesh.addAttrib( Data::Mesh::getAttribName( Data::Mesh::VERTEX_COLOR ), data->getColors() );
+    }
+
+    // To be discussed: Should not weights be part of the geometry ?
+    //        mesh->addData( Data::Mesh::VERTEX_WEIGHTS, meshData.weights );
+
+    mesh.setIndices( std::move( indices ) );
+
+    return mesh;
+}
+
+/// \internal but not so internal ;) what to do ...
```

Comment:
Useless doc. Express first what is the role of the defined element, and then, if you have remarks, questions or ... write them so that they can be understood by others.